dev/com.ibm.ws.cdi.internal/src/com/ibm/ws/cdi/internal/interfaces/ExtensionArchiveProvider.java

```diff
@@ -29,6 +38,11 @@ public interface ExtensionArchiveProvider {
      * @return a collection of additional extension archives to add
      * @throws CDIException if an unexpected exception is encountered
      */
-    Collection<ExtensionArchive> getArchives(CDIRuntime cdiRuntime, WebSphereCDIDeployment deployment) throws CDIException;
+    default Collection<ExtensionArchive> getDiscoveredArchives(CDIRuntime cdiRuntime, WebSphereCDIDeployment deployment) throws CDIException {
+        return Collections.emptySet();
+    }
+
+    ExtensionArchive newSPIExtensionArchive(CDIRuntime cdiRuntime, ServiceReference<CDIExtensionMetadata> sr,
+                                                    CDIExtensionMetadata webSphereCDIExtensionMetaData, WebSphereCDIDeployment applicationContext) throws CDIException;
```

Comment:
This method doesn't make sense on this class. Please put it on its own interface.

Reply:
Do we really need two interfaces for version specific code for generating extension archives? Perhaps rather than splitting the interface we should find a framing where it makes sense to have both methods?

Reply:
As discussed, I think we do, because we need exactly one implementation of this method, whereas we could have any number of the other method.